paimon-core/src/main/java/org/apache/paimon/io/AbstractFileRecordIterator.java

```diff
@@ -32,11 +34,17 @@
  * @param <V> the row type.
  */
 public abstract class AbstractFileRecordIterator<V> implements RecordReader.RecordIterator<V> {
+
+    @Nullable private final PartitionSettedRow partitionSettedRow;
     @Nullable private final ProjectedRow projectedRow;
     @Nullable private final CastedRow castedRow;
 
     protected AbstractFileRecordIterator(
```

Comment:
We need to refine this Iterator.
For example, use `RecordIterator.transform`.

Reply:
Fixed this and removed class `AbstractFileRecordIterator`